ofrak_components/ofrak_components/flash.py

```diff
@@ -314,8 +314,18 @@ async def unpack(self, resource: Resource, config=None):
             magic = flash_attr.ecc_attributes.ecc_magic
             if magic is not None:
                 ecc_magic_offset = data.find(magic)
-                if ecc_magic_offset != -1:
-                    start_index = ecc_magic_offset
+
+                if ecc_magic_offset == -1:
+                    raise UnpackerError("No header magic found")
+
+                search_offset_in_block = flash_attr.get_field_range_in_block(
```

Comment:
Shouldn't this variable be named `field_range_in_block`?

Then line 382 would be `field_range_in_block.start`?